packages/amplify-category-analytics/src/index.ts

```diff
@@ -40,18 +45,24 @@ async function console(context) {
     default:
       break;
   }
-}
+};
 
-async function getPermissionPolicies(context, resourceOpsMapping) {
+/**
+ * Get Permission policies for CloudFormation
+ * @param context cli context
+ * @param resourceOpsMapping - get permission policies for each analytics resource
+ */
+const getPermissionPolicies = async (context: $TSContext, resourceOpsMapping: { [x: string]: $TSAny; }): Promise<$TSAny> => {
   const amplifyMetaFilePath = context.amplify.pathManager.getAmplifyMetaFilePath();
   const amplifyMeta = context.amplify.readJsonFile(amplifyMetaFilePath);
-  const permissionPolicies = [];
-  const resourceAttributes = [];
+  const permissionPolicies: $TSAny[] = [];
+  const resourceAttributes: $TSAny[] = [];
 
   Object.keys(resourceOpsMapping).forEach(resourceName => {
     try {
       const providerName = amplifyMeta[category][resourceName].providerPlugin;
       if (providerName) {
+        // eslint-disable-next-line import/no-dynamic-require, global-require, @typescript-eslint/no-var-requires
         const providerController = require(`./provider-utils/${providerName}/index`);
```

Comment:
here (and other places like this) is this by design? I.e. should we fix these linter warnings in the feature or not?

Perhaps we should have "ProviderProvider/ProviderFactory/ProviderLoader" and centralize this logic?

This is something that can be explored afterwards since this seems to be existing pattern.

Reply:
Like the idea of either have a design where dynamic import is not needed or use centralized factory of some kind if the first isn't possible, but will leave it for now as might increase the scope of this particular move from JS to TS